src/OrchardCore.Modules/OrchardCore.Users/Extensions/UserTimeZoneServiceExtensions.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Modules;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.TimeZone.Services;
+
+namespace OrchardCore.Users;
+
+/// <summary>
+/// Provides an extension for <see cref="IUserTimeZoneService"/>.
+/// </summary>
+public static class UserTimeZoneServiceExtensions
+{
+    /// <summary>
+    /// Gets the time zone for currently logged-in user.
+    /// </summary>
+    /// <param name="userTimeZoneService">The <see cref="IUserTimeZoneService"/>.</param>
+    public static async ValueTask<ITimeZone> GetTimeZoneAsync(this IUserTimeZoneService userTimeZoneService)
+    {
+        ArgumentNullException.ThrowIfNull(nameof(userTimeZoneService));
+
+        var currentUser = await GetCurrentUserAsync(userTimeZoneService.HttpContext);
+
+        return await userTimeZoneService.GetTimeZoneAsync(currentUser);
+    }
+
+    /// <summary>
+    /// Updates the time zone for currently logged-in user.
+    /// </summary>
+    /// <param name="userTimeZoneService">The <see cref="IUserTimeZoneService"/>.</param>
+    public static async ValueTask UpdateTimeZoneAsync(this IUserTimeZoneService userTimeZoneService)
```

Comment:
UpdateCurrentUserAsync(this IUserTimeZoneService userTimeZoneService, HttpContext httpContext)